plugin/payroll/config.go

```diff
@@ -0,0 +1,57 @@
+// plugin/payroll/config.go
+package payroll
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+type PluginConfig struct {
+	Type    string `yaml:"type"`
+	Version string `yaml:"version"`
+	RpcURL  string `yaml:"rpc_url"`
+	Gas     struct {
+		LimitMultiplier int `yaml:"limit_multiplier"`
+		PriceMultiplier int `yaml:"price_multiplier"`
+	} `yaml:"gas"`
+	Monitoring struct {
+		TimeoutMinutes       int `yaml:"timeout_minutes"`
+		CheckIntervalSeconds int `yaml:"check_interval_seconds"`
+	} `yaml:"monitoring"`
+}
+
+func (c *PluginConfig) Validate() error {
+	if c.Type != PLUGIN_TYPE {
+		return fmt.Errorf("invalid plugin type: %s", c.Type)
+	}
+	if c.RpcURL == "" {
+		return errors.New("rpc_url is required")
+	}
+	if c.Gas.LimitMultiplier <= 0 {
+		return errors.New("gas limit multiplier must be positive")
+	}
+	return nil
+}
+
+func loadPluginConfig() (*PluginConfig, error) {
+	configPath := filepath.Join("plugin", "payroll", "payroll.yml")
```

Comment:
We shouldn't hardcode paths - this should look at common unix paths like `/etc/vultisig`, and also support a plugin path flag/config option that's passed down from the main config

Reply:
check now